### PR TITLE
Inline Import Diffs

### PR DESCRIPTION
**Pull request by Astralchroma**
Reopen of #8200 but rebased for master.

Inlines a bunch of import diffs which create merge conflicts and unneeded patch diffs.

### CODE REVIEW

File: patches/api/0021-Add-BaseComponent-sendMessage-methods-to-CommandSend.patch [hunk 1]
```diff
@@ -5,20 +5,10 @@ Subject: [PATCH] Add BaseComponent sendMessage methods to CommandSender
...
+index ec97093a4410322ad44e111d894a3b1a5c5f7b88..b105edf4e907d553848101fef875e8ccd4253ac3 100644
 --- a/src/main/java/org/bukkit/command/CommandSender.java
 +++ b/src/main/java/org/bukkit/command/CommandSender.java
-@@ -1,6 +1,9 @@
- package org.bukkit.command;
- 
- import java.util.UUID;
-+import net.kyori.adventure.audience.MessageType;
-+import net.kyori.adventure.identity.Identity;
-+import net.kyori.adventure.text.Component;
```
Comment (JRoy) [inline patches/api/0021-Add-BaseComponent-sendMessage-methods-to-CommandSend.patch]: Do not inline adventure imports

File: patches/api/0026-Use-ASM-for-event-executors.patch [hunk 2]
```diff
@@ -286,38 +286,19 @@ index 0000000000000000000000000000000000000000..ac99477e9f2c08041aeff31abc1d1ede
...
 diff --git a/src/main/java/org/bukkit/plugin/EventExecutor.java b/src/main/java/org/bukkit/plugin/EventExecutor.java
-index a850f0780de05463fc0d3f9e15ff7f19d88b2aed..9026e108ccd3a88aee1267ee275137befa646455 100644
+index a850f0780de05463fc0d3f9e15ff7f19d88b2aed..0c9ffc9e5385a798b1d4ac486f37806f9e1e5924 100644
 --- a/src/main/java/org/bukkit/plugin/EventExecutor.java
 +++ b/src/main/java/org/bukkit/plugin/EventExecutor.java
-@@ -5,9 +5,75 @@ import org.bukkit.event.EventException;
- import org.bukkit.event.Listener;
- import org.jetbrains.annotations.NotNull;
- 
-+// Paper start
```
Comment (JRoy) [inline patches/api/0026-Use-ASM-for-event-executors.patch]: Imports in this patch are intentional

File: patches/api/0045-Add-String-based-Action-Bar-API.patch [hunk 3]
```diff
@@ -5,18 +5,10 @@ Subject: [PATCH] Add String based Action Bar API
...
 diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
-index e334e7147098169c78f972acbf88757821baa790..17c56b7d4dcac0738078101203fbef1716ed3157 100644
+index 464ba8bb8bff455098010958c8777ef55c844919..ceb56d4a46d4c53ae866d508cede0a9e15c19a5c 100644
 --- a/src/main/java/org/bukkit/entity/Player.java
 +++ b/src/main/java/org/bukkit/entity/Player.java
-@@ -3,6 +3,7 @@ package org.bukkit.entity;
- import java.net.InetSocketAddress;
- import java.util.UUID;
- import com.destroystokyo.paper.Title; // Paper
-+import net.kyori.adventure.text.Component;
```
Comment (JRoy) [inline patches/api/0045-Add-String-based-Action-Bar-API.patch:15]: ditto

File: patches/api/0112-Add-getNearbyXXX-methods-to-Location.patch [hunk 4]
```diff
@@ -5,26 +5,10 @@ Subject: [PATCH] Add "getNearbyXXX" methods to Location
...
-index a90010fea189c5ac9a59a0e6d04c0457243a3280..bbc636baef2e2b0586c7d517be428438ca26ab66 100644
+index a90010fea189c5ac9a59a0e6d04c0457243a3280..710e11effd8ff55f52f832d1f2d707d49e7c1f75 100644
 --- a/src/main/java/org/bukkit/Location.java
 +++ b/src/main/java/org/bukkit/Location.java
-@@ -12,6 +12,15 @@ import org.bukkit.util.Vector;
- import org.jetbrains.annotations.NotNull;
- import org.jetbrains.annotations.Nullable;
- 
-+// Paper start
-+import java.util.Collection;
```
Comment (JRoy) [inline patches/api/0112-Add-getNearbyXXX-methods-to-Location.patch]: Imports in this patch are intentional

File: patches/api/0210-Support-components-in-ItemMeta.patch [hunk 5]
```diff
@@ -5,18 +5,10 @@ Subject: [PATCH] Support components in ItemMeta
...
 diff --git a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
-index 1beedb446a9dd554d05d1d94dba8598e4b69eba6..c6b0fd783675cd019048e445d8e959637d90be7a 100644
+index 1beedb446a9dd554d05d1d94dba8598e4b69eba6..2154fe450f2f26973f67d17cba20d684dfd3692d 100644
 --- a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
 +++ b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
-@@ -5,6 +5,7 @@ import java.util.Collection;
- import java.util.List;
- import java.util.Map;
- import java.util.Set;
-+import net.kyori.adventure.text.Component;
```
Comment (JRoy) [inline patches/api/0210-Support-components-in-ItemMeta.patch:15]: ditto

File: patches/api/0280-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch [hunk 6]
```diff
@@ -6,36 +6,16 @@ Subject: [PATCH] Allow for Component suggestion tooltips in
...
+index a135a9bfb2ccc8842baa9d5760fa05b7b1a529b1..d22e86460e4332e19908690662e2d1c98cb394fc 100644
 --- a/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
 +++ b/src/main/java/com/destroystokyo/paper/event/server/AsyncTabCompleteEvent.java
-@@ -24,6 +24,11 @@
- package com.destroystokyo.paper.event.server;
- 
- import com.google.common.base.Preconditions;
-+import io.papermc.paper.util.TransformingRandomAccessList;
-+import net.kyori.adventure.text.Component;
-+import net.kyori.examination.Examinable;
```
Comment (JRoy) [inline patches/api/0280-Allow-for-Component-suggestion-tooltips-in-AsyncTabC.patch]: Imports in this patch are intentional

File: patches/api/0292-Add-Adventure-message-to-PlayerAdvancementDoneEvent.patch [hunk 7]
```diff
@@ -5,39 +5,28 @@ Subject: [PATCH] Add Adventure message to PlayerAdvancementDoneEvent
...
+index 21ff095afb434d15babcdbe85fa0b94cbb7e75a7..953c9f4d92018b9ab98a1eb6b01e3c6f38091cee 100644
 --- a/src/main/java/org/bukkit/event/player/PlayerAdvancementDoneEvent.java
 +++ b/src/main/java/org/bukkit/event/player/PlayerAdvancementDoneEvent.java
-@@ -4,6 +4,10 @@ import org.bukkit.advancement.Advancement;
- import org.bukkit.entity.Player;
- import org.bukkit.event.HandlerList;
- import org.jetbrains.annotations.NotNull;
-+// Paper start
-+import org.jetbrains.annotations.Nullable;
-+import net.kyori.adventure.text.Component;
```
Comment (JRoy) [inline patches/api/0292-Add-Adventure-message-to-PlayerAdvancementDoneEvent.patch]: ditto

File: patches/api/0328-Add-missing-team-sidebar-display-slots.patch [hunk 8]
```diff
@@ -5,14 +5,10 @@ Subject: [PATCH] Add missing team sidebar display slots
...
 
 diff --git a/src/main/java/org/bukkit/scoreboard/DisplaySlot.java b/src/main/java/org/bukkit/scoreboard/DisplaySlot.java
-index 5d58a18b3625fd01ea34969200edc3bc80cbb587..fe7d0a19f970ac5b4e0c4bef4ff7c4ceae60bb86 100644
+index 5d58a18b3625fd01ea34969200edc3bc80cbb587..227aea4de1160e1fd9539b075aed3c1b6f8907b0 100644
 --- a/src/main/java/org/bukkit/scoreboard/DisplaySlot.java
 +++ b/src/main/java/org/bukkit/scoreboard/DisplaySlot.java
-@@ -1,10 +1,55 @@
- package org.bukkit.scoreboard;
- 
-+import net.kyori.adventure.text.format.NamedTextColor; // Paper
```
Comment (JRoy) [inline patches/api/0328-Add-missing-team-sidebar-display-slots.patch]: ditto

**Review by JRoy — CHANGES_REQUESTED** [review]

**Comment by Astralchroma** [comment]
The issues @JRoy previously mentioned are resolved, I forgot to mention it originally. Also the PR has been rebased. I haven't worked on the server patches yet because the chances of it being accepted are still uncertain, so if that could be clarified that would be great.

**Comment by aurorasmiles** [comment]
This PR will realistically not be merged. While we appreciate it, the sheer scope of it is way too big to just casually merge it, and if we don't merge it now the rebasing will get worse every day. Merging this PR would force every other PR to be rebased, and that isn't something we really want to do outside of big changes like version updates.
Something like this might be done by us during an update in the future, but as other people already mentioned, it just isn't something that's that big of a deal. I'd prefer that things like that either happen in one sweep from us, or just in small chunks during other PRs. For example if you open a PR to fix something in a patch, removing the imports from that patch at the same time would be a good way to do it.

I'll close this now, still thank you for the work!

**Pull request closed without merging**